test/test_nn.py

```diff
@@ -1772,6 +1772,12 @@ def test_weight_norm(self):
         m = nn.Linear(4, 5, dtype=torch.float16)
         m = torch.nn.utils.weight_norm(m)
 
+        m = nn.Linear(4, 5, dtype=torch.bfloat16, device="cpu")
```

Comment:
hm I think instead we should move `test_weight_norm` to  this class https://github.com/pytorch/pytorch/blob/2daf300b5e4489263261b4b943c5aca4a4e1da9c/test/test_nn.py#L7828

The implication here is that the test will be parametrized over device, and then we can pass device=device when we create the `nn.Linear`, you can look at other examples within the class for inspiration.